packages/devtools_app/lib/src/tree.dart

```diff
@@ -63,10 +63,8 @@ class _TreeViewState<T extends TreeNode<T>> extends State<TreeView<T>>
   @override
   void didUpdateWidget(TreeView<T> oldWidget) {
     super.didUpdateWidget(oldWidget);
-    if (widget.dataRoots != oldWidget.dataRoots) {
```

Comment:
I don't think this check was ever `true`. It seems like the `oldWidget.dataRoots` were always identical to the `widget.dataRoots`, regardless of whether the `dataRoots` had changed or not. This meant the tree view wasn't being re-drawn on refresh.

Reply:
I looked into this for the uses of `TreeView` throughout devtools. I think we should change TreeView to accept `dataRootsListenable` instead of `dataRoots`. Then all the `ValueListenableBuilders` around `TreeView` could be removed, and `TreeView` would be responsible for listening to changes in `dataRootsListenable` and calling `setState`. The one place where we aren't using a `List<T>` listenable for the tree view is `ExpandableVariable` (where the listenable has type `T` not a `List<T>`). For this case you can just pass in the raw variable to `ExpandableVariable` and use a `FixedValueListenable` with this value for the `TreeView`

Reply:
Also as a cleanup as part of this change, looks like the VariablesList class can be deleted

Reply:
Done, thanks! Also deleted VariablesList